scio-core/src/main/scala/com/spotify/scio/schemas/Schema.scala

```diff
@@ -117,14 +102,52 @@ object Schema extends JodaInstances with AvroInstances with LowPriorityFallbackI
     ScalaInstances.mapSchema
   implicit def mutableMapSchema[K: Schema, V: Schema]: Schema[mutable.Map[K, V]] =
     ScalaInstances.mutableMapSchema
+
+  final def logicalType[T, U](
+    underlying: BSchema.FieldType,
+    toBase: T => U,
+    fromBase: U => T
+  ): Schema[T] =
+    LogicalType[T, U](underlying, toBase, fromBase)
 }
 
-sealed trait Schema[T] {
+sealed trait Schema[T] extends Serializable {
```

Comment:
wondering why we need to make Serializable now? Do you see some use cases where we need to pass it around?

With this we seem to `ClosureClean` in the `LogicalType` but if we go forward with this should we clean also under `Record` and friends?

Reply:
bc. `SchemaCoder` somehow keeps a reference on `LogicalType` somehow. I'll try to figure out why.

Reply:
So I think I understand what happens. 

When you materialize a `LogicalType`, the beam Schema itself poses no problem BUT the `fromRow` and `toRow` method need to delegate conversions to `toBase` and `fromBase`. So when materializing a logical type, you need to keep a reference on those, and therefore, `LogicalType` needs to be `Serializable`.

Technically you don't need other `Schema` implementations to be `Serializable` but it's not really a problem to mark all of them `Serializable` since they are case classes anyway.

Reply:
hmm interesting 🤔